### ISSUE TITLE
Extending the Collection Day

### ISSUE THREAD

**Issue by teamRibcage** [bruxy70/Garbage-Collection#424]
### Before you submit a new bug report, please check that

- [X] I have read [frequently asked questions and limitations](https://github.com/bruxy70/Garbage-Collection/wiki/Review-before-submitting-a-new-issue-or-feature-request)

### Write your question here

I have a bit of a unique situation in which I would like to track a collection that occurs sometime during the first full week of the month. I have no issue scheduling collection for the 1st Monday of the month (and therefore the 1st full business week), however I would like the entity to retain the "Today" (or 0) status through Friday. I have so far been unsuccessful in configuring such a scheme - any help would be greatly appreciated!

Thanks! 🙂

**Comment by bruxy70** [comment]
I suggest creating another sensor, a template, checking whether the difference between current date and the last collection date (this is attribute of the collection) is less or equal than 5. And you can make it either a sensor, a binary sensor (yes or no), or even evaluate this as a condition in your automation and not create any new sensor, just use the last collection date and check whether  it was less than 5 days ago.

**Comment by teamRibcage** [comment]
I was actually just thinking of a Template Sensor as well! I will explore a few different options in that direction.....

Thanks for your quick response and all your hard work on this Integration/Helper!!
🙂

**Comment by teamRibcage** [comment]
Well, I'm happy to report that I have it working exactly as I had hoped it would!

I ended up creating two sensors that run behind the scenes to define the Start and End dates - this allows me to make parameters specific to each sensor, as well as easily change them later if needed. The Start sensor resets when the day changes, the End sensor resets at noon of its collection day. I then created a template sensor combining the two to use in the front end of HomeAssistant. End result, Heavy Pickup Week begins on Monday of the second full week of the month and lasts through noon on Friday.

![heavy_pickup](https://user-images.githubusercontent.com/76419115/191348762-62be64ab-0c34-48e4-aa59-4cd24d422337.jpg)

```
template:
  - sensor
    - name: Heavy Pickup Week
      icon: mdi:dump-truck
      unique_id: heavypickupweek
      state: |
        {% if state_attr('sensor.heavy_pickup_start', 'days') == 1 %}
          Tomorrow
        {% elif state_attr('sensor.heavy_pickup_end', 'days') <= 4 %}
          This Week
        {% elif state_attr('sensor.heavy_pickup_start', 'days') >= 2 %}
          {{ states('sensor.heavy_pickup_start') }}
        {% else %}
          Calculating
        {% endif %}
```

Thanks again for your help!! 🙂
